core/src/main/java/org/apache/druid/data/input/impl/MapInputRowParser.java

```diff
@@ -93,23 +94,32 @@ public static InputRow parse(
     final DateTime timestamp;
     try {
       timestamp = timestampSpec.extractTimestamp(theMap);
-      if (timestamp == null) {
-        final String input = theMap.toString();
-        throw new NullPointerException(
-            StringUtils.format(
-                "Null timestamp in input: %s",
-                input.length() < 100 ? input : input.substring(0, 100) + "..."
-            )
-        );
-      }
     }
     catch (Exception e) {
-      throw new ParseException(e, "Unparseable timestamp found! Event: %s", theMap);
+      throw new ParseException(e, "Unparseable timestamp found! Event: %s", rawMapToPrint(theMap));
+    }
+    if (timestamp == null) {
+      throw new ParseException("Unparseable timestamp found! Event: %s", rawMapToPrint(theMap));
+    }
+    if (!Intervals.ETERNITY.contains(timestamp)) {
+      throw new ParseException(
+          "Encountered row with timestamp that cannot be represented as a long: [%s]",
+          rawMapToPrint(theMap)
+      );
```

Comment:
After your refactoring in this change, the `parse` functions on line 65 and 70 can be made private and package private (VisibleForTesting) respectively

Reply:
Done.